File: dotnet/private/tests/resx/Greeter.fs
module Greeter

open System.Globalization
open System.IO
open System.Reflection
open System.Resources

let private strings =
    ResourceManager("ResxFsTest.Strings", Assembly.GetExecutingAssembly())

// This resource uses an explicit `logical_names` override, so its manifest name is
// independent of the `.resx` file path and the assembly's root namespace.
let private messages =
    ResourceManager("MyApp.Messages", Assembly.GetExecutingAssembly())

let private get (rm: ResourceManager) (key: string) (culture: string) =
    match rm.GetString(key, CultureInfo.GetCultureInfo(culture)) with
    | null -> "<null>"
    | s -> s

[<EntryPoint>]
let main argv =
    // Force '\n' so the output matches on every platform.
    use writer = new StreamWriter(argv.[0])
    writer.NewLine <- "\n"
    writer.WriteLine("neutral: " + get strings "Hello" "")
    writer.WriteLine("fr: " + get strings "Hello" "fr")
    writer.WriteLine("messages neutral: " + get messages "Greeting" "")
    writer.WriteLine("messages fr: " + get messages "Greeting" "fr")
    0

File: dotnet/private/tests/resx/Greeter.cs
using System;
using System.Globalization;
using System.IO;
using System.Resources;

namespace ResxCsTest
{
    public static class Greeter
    {
        private static readonly ResourceManager Strings =
            new ResourceManager("ResxCsTest.Strings", typeof(Greeter).Assembly);

        // This resource uses an explicit `logical_names` override, so its manifest name is
        // independent of the `.resx` file path and the assembly's root namespace.
        private static readonly ResourceManager Messages =
            new ResourceManager("MyApp.Messages", typeof(Greeter).Assembly);

        private static string Get(ResourceManager rm, string key, string culture) =>
            rm.GetString(key, CultureInfo.GetCultureInfo(culture)) ?? "<null>";

        public static void Main(string[] args)
        {
            // Force '\n' so the output matches on every platform.
            using var writer = new StreamWriter(args[0]) { NewLine = "\n" };
            writer.WriteLine("neutral: " + Get(Strings, "Hello", ""));
            writer.WriteLine("fr: " + Get(Strings, "Hello", "fr"));
            writer.WriteLine("messages neutral: " + Get(Messages, "Greeting", ""));
            writer.WriteLine("messages fr: " + Get(Messages, "Greeting", "fr"));
        }
    }
}
